### PR TITLE
DOCK-2349: added error handling message for CWL JSON conversion

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
Currently, generating test parameter JSON files for CWL workflows with arrays for `items` ([example](https://qa.dockstore.org/workflows/github.com/bcbio/bcbio_validation_workflows/wes-agha-test-gcp:master?tab=files)) results in a `ClassCastException` error due to these [lines](https://github.com/dockstore/cwlavro/blob/83e0bd298060472bea42d638460c5899a67380d0/cwlavro-tools/src/main/java/io/cwl/avro/CWL.java#L307-L308).

This PR adds an error handling message to let users know that it's file structure is not supported yet. 

Before fix: 
```
dockstore workflow convert entry2json --entry github.com/bcbio/bcbio_validation_workflows/wes-agha-test-gcp:master
Current version : 1.13.1
You have the most recent stable release.
If you wish to upgrade to the latest unstable version, please use the following command:
   dockstore --upgrade-unstable
14:37:29.011 [main] ERROR io.dockstore.client.cli.ArgumentUtility - java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.lang.String (java.util.ArrayList and java.lang.String are in module java.base of loader 'bootstrap')
	at io.cwl.avro.CWL.getStub(CWL.java:307)
	at io.cwl.avro.CWL.extractRunJson(CWL.java:138)
	at io.github.collaboratory.cwl.CWLClient.generateInputJson(CWLClient.java:411)
	at io.dockstore.client.cli.nested.WorkflowClient.convertWorkflow2Json(WorkflowClient.java:289)
	at io.dockstore.client.cli.nested.WorkflowClient.handleEntry2json(WorkflowClient.java:273)
	at io.dockstore.client.cli.nested.AbstractEntryClient.convert(AbstractEntryClient.java:644)
	at io.dockstore.client.cli.nested.AbstractEntryClient.processEntryCommands(AbstractEntryClient.java:309)
	at io.dockstore.client.cli.Client.run(Client.java:737)
	at io.dockstore.client.cli.Client.main(Client.java:633)
```

After fix:
```
dockstore workflow convert entry2json --entry github.com/bcbio/bcbio_validation_workflows/wes-agha-test-gcp:master
ERROR: Cannot generate a JSON for this workflow.
The structure of its input is currently not supported. Please provide items as a String.
```
**Review Instructions**
Review that the new error message is shown.

**Issue**
https://github.com/dockstore/dockstore/issues/5418

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `./mvnw clean install` 
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Review by coverbeck — COMMENTED** [review]
Shouldn't we fix this in CWL Avro?

**Comment by denis-yuen** [comment]
> Shouldn't we fix this in CWL Avro?

I suggested a small fix for 1.14 and fixing in cwl avro in 1.15 (assuming we don't make another attempt at cwljava)

**Review by denis-yuen — CHANGES_REQUESTED** [review]
That said, this PR is to develop when it should be to the release branch

**Comment by hyunnaye** [comment]
> That said, this PR is to develop when it should be to the release branch

fixed - sorry about that

File: dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java
Comment (coverbeck) [inline dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java:455]: Is the workaround to have:

```
type: string[]
```

instead of 

```
type: array
items: string
```

If yes, your message could be a little more specific. Also, I think (?) CWL is case-sensitive; if so, you should have `string` (`string[]`?) instead of `String`.

**Review by coverbeck — COMMENTED** [review]
> I suggested a small fix for 1.14 and fixing in cwl avro in 1.15 (assuming we don't make another attempt at cwljava)

Is there a ticket for 1.15?

I left a nit-picky comment that may not be worth the effort for this bug, but waiting for build issue to be sorted out before approval.

File: dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java
Comment (hyunnaye) [reply dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java]: Due to the [casting](https://github.com/dockstore/cwlavro/blob/83e0bd298060472bea42d638460c5899a67380d0/cwlavro-tools/src/main/java/io/cwl/avro/CWL.java#L307-L308) here, both `type` and `items` should be strings. I edited the message to say `string` instead of `String`

**Comment by hyunnaye** [comment]
1.15 ticket created here https://github.com/dockstore/dockstore/issues/5508

**Pull request merged**
